subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DarwinFileWatcherRegistryFactory.java

```diff
@@ -31,23 +31,23 @@
 import java.util.concurrent.TimeUnit;
 import java.util.function.Predicate;
 
-public class DarwinFileWatcherRegistryFactory extends AbstractFileWatcherRegistryFactory<OsxFileEventFunctions> {
+public class DarwinFileWatcherRegistryFactory extends AbstractFileWatcherRegistryFactory<OsxFileEventFunctions, OsxFileWatcher> {
```

Comment:
```suggestion
public class OsxFileWatcherRegistryFactory extends AbstractFileWatcherRegistryFactory<OsxFileEventFunctions, OsxFileWatcher> {
```

Should we rename this, so the names for macOS align better?

Reply:
I'm trying to keep this change to the minimum, as we need to merge it back to 6.9.1. Let's do this rename separately.

Reply:
👍